### ISSUE TITLE
cauchyintegral

### ISSUE THREAD

**Issue by MikaelSlevinsky** [ApproxFun/SIE.jl#10]
For `Hilbert(d,0)`, the singular term is log|y-x|.

In `cauchyintegral`, it seems this is log(y-x), and the following crashes:

```julia
using ApproxFun, SIE
ds = JacobiWeight(.5,.5,Chebyshev())
f = Fun(x->exp(x)*sqrt(1-x^2),ds)
cauchyintegral(f,2.)
cauchyintegral(f,-2.)
```

It seems to me this will be required if we want to do scattering by a screen, so how to proceed? What is the correct analytic continuation, similar to `sqrtx2`?

**Comment by dlfivefifty** [comment]
I think that cauchyintegral doesn't get the integration constant quite right, I'm in the process of fixing that.  I'll then implement a logkernel directly since it's just real part of cauchyintegral.

This should make clear how to implement Stieljes(0)

**Comment by MikaelSlevinsky** [comment]
Can't we just replace `log|y-x|` with `.5log(abs2(y-x))`?

**Comment by dlfivefifty** [comment]
I've added logabs for this in singfuncauchy

**Comment by MikaelSlevinsky** [comment]
This is great! Thanks again. I wonder if we could feed it a tuple, and at every step extract the real and imaginary parts of horner terms to create the outgoing tuple. I think this would allow one to use logkernel for complex functions in R^2 instead of C.

**Comment by dlfivefifty** [comment]
Wouldn’t this be computationally equivalent to doing e.g.

	logabs(x::(Float64,Float64))=logabs(x[1]+im*x[2])

Since the computation is exactly what complex arithmetic does?

**Comment by MikaelSlevinsky** [comment]
You're right. Replacing the signature `realintegratejin{T<:Real}(cfs::Vector{T},y)` with `realintegratejin(cfs,y)` then allows this:

```julia
using AppoxFun, SIE
z=10.+im
f = Fun(x->exp(im*x)*sqrt(1-x^2),JacobiWeight(.5,.5,Chebyshev()))
quadgk(x->f[x]*log(sqrt((x-z.re)^2+z.im^2)),-1,1;reltol=eps())
logkernel(f,z)
```

it should only take minor wrapping for the tuples.

Is the integration constant right for inverse square roots yet?

```julia
f = Fun(x->exp(x)/sqrt(1-x^2),JacobiWeight(-.5,-.5,Chebyshev()))
quadgk(x->f[x]*log(sqrt((x-z.re)^2+z.im^2)),-1,1)
logkernel(f,z)
```

**Comment by dlfivefifty** [comment]
It is now :)  (another off by log(2) issue)

x=Fun(identity)
f=exp(x)/sqrt(1-x^2)
z=1.+1.im;
logkernel(f,z)-sum(f*log(abs(z-x)))

**Comment by MikaelSlevinsky** [comment]
I think there's a complex bug in sum:

```julia
x=Fun(identity)
f=exp(im*x)/sqrt(1-x^2)
z=1.+1.im;
logkernel(f,z)-sum(f*log(abs(z-x)))
logkernel(f,z)
quadgk(x->f[x]*log(abs(z-x)),-1,1)
```

**Comment by MikaelSlevinsky** [comment]
complex bug in sum fixed @1986ca652ae89106d6502a5f5e1080f9b9908a5f.
